src/librustdoc/config.rs

```diff
@@ -374,6 +377,31 @@ impl Options {
             }
         };
 
+        let mut default_settings: Vec<Vec<(String, String)>> = vec![
+            matches
+                .opt_str("default-theme")
+                .iter()
+                .map(|theme| {
+                    vec![
+                        ("use-system-theme".to_string(), "false".to_string()),
+                        ("theme".to_string(), theme.to_string()),
+                    ]
+                })
+                .flatten()
+                .collect(),
+            matches
+                .opt_strs("default-setting")
+                .iter()
+                .map(|s| {
+                    let mut kv = s.splitn(2, '=');
+                    let k = kv.next().unwrap().to_string();
```

Comment:
```suggestion
                    // never panics because `splitn` always returns at least one element
                    let k = kv.next().unwrap().to_string();
```